Changes

darksmo opened the pull request:
Hello, I've been using djax for a project and made a bunch of changes.

See commit messages for details.

I was wondering if you would like to pull and merge the changes in the main repository.

Changes are basically:
- ajaxLoaded is an additional event triggered when all the items have been replaced
- added one more rule for replacing blocks of markup (see 58f03a0)
- all click events are namespaced
- cross-browser title replacement (wasn't working on IE7)
- public navigate method to trigger navigation programmatically
- more standard jquery plugin interface (see 53871aa)

Cheers,
Savio

darksmo committed:
replace title cross-browser on ajax update

darksmo committed:
trigger event once all markup is replaced

darksmo committed:
use $.ajax call instead of $.get call

darksmo committed:
append new block to previous block if no parent

1) in case no previous sibling of a certain *new* block is
   found, djax used to prepend this block within its parent,
   and this behaviour is preserved.

2) however, in case a new block has no parent (or no id is
   specified for the parent enclosing a certain set of
   blocks) the new block is inserted after the previous
   block.

The case 2) is triggered with a response like:

  <div>
    <div id="...">...</div>
    <div id="...">...</div>
    <div id="...">...</div>
  </div>

  (note that the enclosing block has no id).

darksmo committed:
namespace for click events

darksmo committed:
code refactoring

- public method to trigger a djax update programmatically
- make code more standard as explained in http://learn.jquery.com/plugins/basic-plugin-creation/

darksmo committed:
update documentation

darksmo committed:
rename variable for readability

darksmo committed:
add public navigate method

darksmo committed:
store djaxing status in globally

beezee commented:
@darksmo all sounds great and thanks for opening- this will take me some time to evaluate since the diff is so large.

One note from initial readthrough- the syntax is incorrect in your updated readme examples (missing curly brackets around parameter object)

Other than that I'll aim to give a line by line readthrough within the next week and I'll add any comments to specific commits as necessary, or merge if everything looks good.

Thanks again for all this, very cool pull request.

darksmo committed:
fix examples syntax in readme.md

darksmo commented:
Cool, have fixed the syntax in the documentation...

darksmo committed:
handle user navigating quickly

- only the last click is finally considered the intended click.
- if the user clicks too fast, we assume he doesn't want to even store the intermediate locations in the history...

darksmo committed:
enable cross-domain requests on IE

darksmo committed:
some error handling

darksmo committed:
better naming and replacement logic

darksmo committed:
get ajax settings from initialiization

darksmo committed:
Merge branch 'master' of https://github.com/darksmo/djax

darksmo committed:
support blocking callback on djax click

darksmo committed:
deferred djax

Commit:
Support request method and parameters

Commit:
Deal with the targetUrl header on errors

Commit:
Wrap reponse to obtain all the descendants

Commit:
Add ajax timeout time and callback to settings

darksmo committed:
Merge pull request #1 from sbscully/master

Add ajax timeout time and callback to settings

WTVader commented:
Recently I came across your fork of djax and tried it on a project. I was wondering if the script is working properly since I got multiple errors due to e.g. the following code:

```
$('a:not(.' + settings.ignoreClass + ')',…
```

But even by changing the code to get rid of the errors the script doesn’t insert anything from the ajax call. Maybe you can help me to get the script working as desired.

darksmo commented:
Hello @WTVader, please ignore that fork.

After [this commit](https://github.com/darksmo/djax/commit/d77594561b48fdbe5480bfdf6a9c491455487641) it really diverges from the original project (DJAX). I should perhaps delete it or point it in the README to avoid further confusion around.

The one I'm currently using is DDJAX (see https://github.com/lokku/jquery-ddjax). It shares the same idea as DJAX, but the implementation of block replacements is left to you (e.g., for better performance).

darksmo committed:
Update readme.md
